marlowe/security.md

```diff
@@ -0,0 +1,112 @@
+# Marlowe Security Guide
+
+- [Contracts](#contracts)
+- [Transactions](#transactions)
+- [Marlowe Runtime](#marlowe-runtime)
+- [Marlowe CLI](#marlowe-cli)
+
+
+## Contracts
+
+See the [Best Practices for Marlowe on Cardano](./best-practices.md) for guidelines on verifying that a Marlowe contract will execute correctly on the Cardano blockchain. Use Marlowe Playground or `marlowe-cli run prepare` to simulate the execution of the contract.  Use the `marlowe-cli run analyze` tool to check for Cardano-specific problems in the contract, or run the contract itself on one of the Cardano testnets. In general, one should not participate in a smart contract whose execution paths one does not understand.
+
+
+## Transactions
+
+The spending of a UTxO for a Marlowe contract or a payout is validated by a Plutus script, either the semantics validator or the payout validator. For maximum security, one should not sign a transaction unless one has reviewed and understood the contents and implications of the unsigned transaction. For Marlowe, this means verifying the Marlowe contract and its input.
+
+
+### Marlowe Semantics Validator
+
+Before signing a Marlowe transaction, one should understand the following:
+1. Does the transaction operate a Marlowe contract?
+2. What is the current contract and its state?
+3. What input is being applied to the contract?
+4. What else is occurring in the transaction?
+
+
+#### Does the transaction operate a Marlowe contract?
```

Comment:
Should we add something about the minting policy? Are there any other things that need considering regarding the initial state of the contract, the tokens being created explicitly for this contract, or anything like that?

Reply:
Yes, I'll add a discussion on minting policy, which is a very important security consideration.

Reply:
Added in 8a6b693d0d03bd91478b2bc4666c268edefdbd7d.